File: ChatApplication.Domain/Entities/Friend.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChatApplication.Domain.Entities
{
    public class Friend : BaseEntity
    {
        public string SenderId { get; set; }

        [ForeignKey("SenderId")]
        public ApplicationUser Sender { get; set; }

        public string ReceiverId { get; set; } = string.Empty;

        [ForeignKey("ReceiverId")]
        public ApplicationUser Receiver { get; set; }

        public FriendStatus Status { get; set; } = FriendStatus.Beklemede;
        public DateTime RequestDate { get; set; } = DateTime.UtcNow;
        public DateTime? AcceptedDate { get; set; }


    }
    public enum FriendStatus
    {
        Beklemede,
        Onaylandi,
        Rededildi,
        Engellendi
    }
}

File: ChatApplicationClient/src/app/core/shared/models/friend.ts
import { User } from "./user";

export type Friend = {
    id: string;
    senderId: string;
    sender?: User;
    receiverId: string;
    receiver?: User;
    status: FriendStatus;
    requestDate: Date;
    acceptedDate?: Date;
    name: string;
    lastName: string;
    email: string;
    userName: string;
    profilePhotoUrl?: string;
    unreadMessageCount?: number;
};

export enum FriendStatus {
    Beklemede = 'Beklemede',
    Onaylandi = 'Onaylandi',
    Rededildi = 'Rededildi',
    Engellendi = 'Engellendi'
}

export interface LastMessage {
  content: string;
  sentAt: Date;
  isRead: boolean;
  senderId: string;
}

export interface PendingFriendRequest {
  friendshipId: string;
  senderId: string;
  senderName: string;
  senderLastName: string;
  senderEmail: string;
  requestDate: string;
  senderProfilePhotoUrl: string | null;

}

